### PR TITLE
Snapshot retention test framework changes

### PR DESCRIPTION
**Pull request by susesgartner**
All framework adjustments needed for the snapshot retention test as well as some updates to how we handle constants across the framework and in rancher tests. 

https://github.com/rancher/qa-tasks/issues/895

### CODE REVIEW

File: extensions/etcdsnapshot/etcdsnapshot.go
Comment (markusewalker) [inline extensions/etcdsnapshot/etcdsnapshot.go:291]: nit - can this variable be named something more clear instead of `msg`?

File: extensions/etcdsnapshot/etcdsnapshot.go
Comment (markusewalker) [inline extensions/etcdsnapshot/etcdsnapshot.go:296]: maybe the first part of `Snapshot retention limit respected` can just be removed?

File: extensions/etcdsnapshot/etcdsnapshot.go
Comment (markusewalker) [inline extensions/etcdsnapshot/etcdsnapshot.go:302]: nit - similar to the RKE1 function, can this and the comment be renamed to something like `v2ProvRetentionLimitCheck`?

File: extensions/etcdsnapshot/etcdsnapshot.go
Comment (susesgartner) [reply extensions/etcdsnapshot/etcdsnapshot.go:253]: Sounds good to me.

File: extensions/etcdsnapshot/etcdsnapshot.go
Comment (susesgartner) [reply extensions/etcdsnapshot/etcdsnapshot.go:272]: Good point.

File: extensions/etcdsnapshot/etcdsnapshot.go
Comment (susesgartner) [reply extensions/etcdsnapshot/etcdsnapshot.go:296]: I don't have strong feelings one way or another on this so i'll change it.

File: extensions/etcdsnapshot/etcdsnapshot.go
Comment (susesgartner) [reply extensions/etcdsnapshot/etcdsnapshot.go:291]: I will change it to errMsg

File: extensions/etcdsnapshot/etcdsnapshot.go
Comment (slickwarren) [inline extensions/etcdsnapshot/etcdsnapshot.go:313]: i think what you have could be okay, but typically we convert the spec to a steve/management object first, then pull the variable from it. i.e. this would probably be 
```
	spec := &provv1.ClusterSpec{}
	err = steveV1.ConvertToK8sType(clusterObj.Spec, spec)
	if errr...
```
instead of type casting through go

File: extensions/provisioning/verify.go
Comment (slickwarren) [inline extensions/provisioning/verify.go:445]: I think it would be worth a comment in the code as to why we expect the list to be longer for rke1

File: extensions/etcdsnapshot/etcdsnapshot.go
Comment (susesgartner) [reply extensions/etcdsnapshot/etcdsnapshot.go:14]: Nope i'll go fix that.

File: extensions/etcdsnapshot/etcdsnapshot.go
Comment (susesgartner) [reply extensions/etcdsnapshot/etcdsnapshot.go:266]: I'll update that. As to if there is a linter check for that I can dig around a bit

File: extensions/etcdsnapshot/etcdsnapshot.go
Comment (susesgartner) [reply extensions/etcdsnapshot/etcdsnapshot.go:272]: Just got my linter working in vscode and it caught this

File: extensions/etcdsnapshot/etcdsnapshot.go
Comment (susesgartner) [reply extensions/etcdsnapshot/etcdsnapshot.go:278]: Great Idea I will  implement something for this. For Norman it looks like I can use Filters. I will hunt around for a way to do this in Steve.

File: extensions/etcdsnapshot/etcdsnapshot.go
Comment (susesgartner) [reply extensions/etcdsnapshot/etcdsnapshot.go:287]: Good catch, i'll change that.

File: extensions/etcdsnapshot/etcdsnapshot.go
Comment (susesgartner) [reply extensions/etcdsnapshot/etcdsnapshot.go:313]: Glad there is an easier way to do this, i'll change it.

**Comment by slickwarren** [comment]
psa; there are some conflicts to resolve too, when you have time.

File: extensions/provisioning/verify.go
Comment (susesgartner) [reply extensions/provisioning/verify.go:445]: I just noticed that logic. I am not entirely sure why it is expected number of snapshots to be higher then the expected amount.

**Comment by susesgartner** [comment]
Fixed the merge conflicts

File: extensions/etcdsnapshot/etcdsnapshot.go
Comment (susesgartner) [reply extensions/etcdsnapshot/etcdsnapshot.go]: After looking into this a bit, i'm going to create a separate ticket for this since I would need to update all references to GETRKE1Snapshots to implement this (scope creep).

**Review by slickwarren — APPROVED** [review]

File: extensions/provisioning/verify.go
Comment (markusewalker) [reply extensions/provisioning/verify.go:445]: That's by design. In the scenario that you are running this against an RKE1 cluster you just provisioned, there is always going to be a snapshot that is kicked off right afterwards. Because of that, the length is going to be +1 then what you expect.

**Review by markusewalker — COMMENTED** [review]
LGTM. Will approve once a rebase occurs. Also, once this is approved, can you open the backport PRs for v2.7 and v2.8?

**Comment by susesgartner** [comment]
Finished rebasing this PR.

**Review by markusewalker — APPROVED** [review]

**Review by Josh-Diamond — APPROVED** [review]

File: extensions/defaults/apiobjects.go
Comment (igomez06) [inline extensions/defaults/apiobjects.go:1]: One last nit Sam. Could put this in a different package, something more descriptive than just defaults?

**Review by igomez06 — COMMENTED** [review]
Also Sam can you rewrite your commit message? Revert[] is not very descriptive either.

File: extensions/defaults/apiobjects.go
Comment (susesgartner) [reply extensions/defaults/apiobjects.go]: Fixed this.

**Review by caliskanugur — APPROVED** [review]
Looks good to me

**Pull request merged**
